src/error.ts

```
@@ -1018,6 +1018,62 @@ export class MongoTopologyClosedError extends MongoAPIError {
   }
 }
 
+/**
+ * An error generated when the MongoClient is closed and async
+ * operations are interrupted.
+ *
+ * @public
+ * @category Error
+ */
+export class MongoClientClosedError extends MongoAPIError {
```

I don't see this class used anywhere.  Dead code?

---

two questions

1. do we expect users to have divergent logic depending on whether or not an operation failed because the client was closed?  if not, then perhaps we don't need a specialized error class for this particular scenario.
2. (if 1 is yes) is MongoAPIError the correct base class?  This seems more like a MongoDriverError, because I don't think we consider closing the client while operations are in progress an "API error".

---

I don't expect users to have divergent logic because, ideally, they do not have outstanding operations to receive this error. However, we have divergent logic the PoolClosedError is related to our retry and resume machinery, which should not be responding to the "close in-use" step of shutdown.

Now, that being said, without an error for client close it may be unclear what went wrong with an operation, PoolClosedErrors happen during the lifetime of our client and ideally are recovered internally. If the user is exposed to one it doesn't mean the end of their MongoClient. However this error does mean that, unless they re-connect the client was closed intentionally, so they know what to look for if this is a source of a bug (ex. some event/timer fires and leads to a client.close() call, when my find() returns `MongoClientClosedError` as a result I know that isn't the same as a monitoring health issue)

That is a fair point about the subclass `MongoAPIError` is a subclass of the `MongoDriverError` already and I used it here because `MongoTopologyClosedError` is already subclassing `MongoAPIError`. I agree it doesn't fit your usual expectation of an API error, but if the ideal is that there aren't pending operations when client.close is reached then it is somewhat of an indication of API misuse to receive this error.

---

I agree that it is important to provide users with an error that conveys the underlying issue.  I don't know that I necessarily agree that we need a new error class to communicate what went wrong to the user though.  Could we just throw a descriptive error using one of our existing classes?  ex: `new MongoAPIError('Operation interrupted because client was closed')`

---

I feel pretty strongly that a descriptive error name is worth having here, along with a meaningful message, so I can use the string you suggest here